Player.gd

```diff
@@ -48,21 +48,55 @@ func get_player_input():
 			$AnimationPlayer.play("Attack")
 			$Networking.sync_character_animation = "Attack"
 			
-			rpc_id(1, "spawn_projectile")
+			rpc_id(1, "spawn_projectile_server")
+
 @rpc(any_peer)
-func spawn_projectile():
+func spawn_projectile_server():
+	# TODO consider adding the path to the attack you want.
+	# That way different attacks could be triggered by just changing one argument :)
+	
+	if not multiplayer.is_server():
```

Comment:
Because this has @rpc(any_peer) any client could call this method on any machine!
So the first thing we should do in _every_ rpc that a client can call is to check whether we're the server.
If not it means something dodgy is happening, like Client A is directly telling Client B to run code!